QRCoder/QRCodeGenerator.cs

```diff
@@ -745,14 +745,22 @@ bool IsUtf8()
             }
         }
 
+        private static readonly Encoding _iso88591ExceptionFallback = Encoding.GetEncoding(28591, new EncoderExceptionFallback(), new DecoderExceptionFallback()); // ISO-8859-1
         /// <summary>
         /// Checks if the given string can be accurately represented and retrieved in ISO-8859-1 encoding.
         /// </summary>
         private static bool IsValidISO(string input)
         {
-            var bytes = Encoding.GetEncoding("ISO-8859-1").GetBytes(input);
-            var result = Encoding.GetEncoding("ISO-8859-1").GetString(bytes);
-            return String.Equals(input, result);
+            // No heap allocations if the string is ISO-8859-1
+            try
+            {
+                _ = _iso88591ExceptionFallback.GetByteCount(input);
+                return true;
+            }
+            catch (EncoderFallbackException) // The exception is a heap allocation and not ideal
```

Comment:
FYI: starting w/ .NET 8 one could use [`Encoding.TryGetBytes`](https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.trygetbytes?view=net-8.0) for this to avoid the slow exception path.

(I know that ATM there's no target for .NET 8 or newer in this project)

Reply:
`Encoding.TryGetBytes`, where available, only is useful if it is determined that the buffer is too small.  So potentially it could eliminate `GetByteCount`, but it will not avoid the exception path.

This method should be eliminated in a separate PR anyway - see #514 .  In this PR the logic is unchanged so the method is necessary.